### PR TITLE
report and register via grpc

### PR DESCRIPTION
**Pull request by majst01**
With this we remove the direct dependency to [metal-core](https://github.com/metal-stack/metal-core) and indirect calls to [metal-api](https://github.com/metal-stack/metal-api). Instead [pixie](https://github.com/metal-stack/pixie) returns in the first call all required info to make direct metal-api calls. With this we also can move `pixie` away from every switch and place it in a central location.

Depends on:

- [x] https://github.com/metal-stack/metal-api/pull/279
- [x] https://github.com/metal-stack/metal-go/pull/83
- [x] https://github.com/metal-stack/pixie/pull/7

TODO:

- [x] testing with install, reinstall, abort-reinstall and different filesystemlayouts
- [x] adopt metal-roles

### CODE REVIEW

**Comment by majst01** [comment]
Only interfaces with neighbors must be added to install.yaml

**Comment by Gerrit91** [comment]
Requires changes for docs.

File: cmd/metal-client.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, metalConfig.GRPCAddress, grpcOpts...)
+	if err != nil {
+		return nil, err
+	}
+	// FIXME set to Metal-View
```
Comment (Gerrit91) [inline cmd/metal-client.go:92]: Can be removed?

**Review by Gerrit91 — APPROVED** [review]

File: cmd/metal-client.go (hunk 1 above)
Comment (majst01) [reply cmd/metal-client.go]: done

**Pull request merged**